### PR TITLE
 refactoring: add a defensive check for nullptr.

### PR DESCRIPTION
**Pull request by nkcsgexi**
rdar://48290050

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

File: lib/IDE/Refactoring.cpp [hunk 1]
```diff
@@ -1909,8 +1909,9 @@ class ExpandableAssignTernaryExprInfo: public ExpandableTernaryExprInfo {
   IfExpr *getIf() {
     if (!Assign)
       return nullptr;
-
-    return dyn_cast<IfExpr>(Assign->getSrc());
+    if (auto *Src = Assign->getSrc())
```
Comment (theblixguy) [inline lib/IDE/Refactoring.cpp]: I think you could just do `return dyn_cast_or_null<IfExpr>(Assign->getSrc);`

Comment (nkcsgexi) [reply lib/IDE/Refactoring.cpp]: Done.

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Pull request merged**